modules/ppcp-wc-gateway/services.php

```diff
@@ -2093,4 +2093,23 @@ static function( ContainerInterface $container ): DisplayManager {
 	'wcgateway.settings.admin-settings-enabled'            => static function( ContainerInterface $container ): bool {
 		return $container->has( 'settings.url' ) && ! SettingsModule::should_use_the_old_ui();
 	},
+
+	/**
+	 * Returns a prefix for the site, ensuring the same site always gets the same prefix (unless the URL changes).
+	 */
+	'wcgateway.settings.invoice-prefix'                    => static function( ContainerInterface $container ): string {
+		$site_url = get_site_url( get_current_blog_id() );
+		$hash     = md5( $site_url );
+		$letters  = preg_replace( '~\d~', '', $hash ) ?? '';
+		$prefix   = substr( $letters, 0, 6 );
+
+		return $prefix ? $prefix . '-' : '';
+	},
+
+	/**
+	 * Returns random 6 characters length alphanumeric prefix, followed by a hyphen.
+	 */
+	'wcgateway.settings.invoice-prefix-random'             => static function( ContainerInterface $container ): string {
+		return wp_generate_password( 6, false ) . '-';
```

Comment:
I guess it should not cause issues for us currently since afaik we don't create multiple orders in a single request, but implementing this as a service feels wrong, since it will cached. Maybe it can be a Factory at least (they are not cached).